src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.Span.cs

```diff
@@ -43,13 +47,17 @@ public static TValue Deserialize<TValue>(ReadOnlySpan<byte> utf8Json, JsonSerial
         /// <param name="returnType">The type of the object to convert to and return.</param>
         /// <param name="options">Options to control the behavior during parsing.</param>
         /// <exception cref="System.ArgumentNullException">
-        /// Thrown if <paramref name="returnType"/> is null.
+        /// <paramref name="returnType"/> is null.
         /// </exception>
         /// <exception cref="JsonException">
         /// Thrown when the JSON is invalid,
         /// <paramref name="returnType"/> is not compatible with the JSON,
         /// or when there is remaining data in the Stream.
         /// </exception>
+        /// <exception cref="NotSupportedException">
+        /// There is no compatible <see cref="System.Text.Json.Serialization.JsonConverter"/>
+        /// for <paramref name="returnType"/>.
```

Comment:
We'd also throw NSE if any of the types within the object graph of `returnType` don't have a `JsonConverter`, not just `returnType`.

Reply:
Yes I considered something like "There is no compatible JsonConverter for {0} or any of its properties" but that extra verbage I thought was implied. A converter includes anything it does including delegating to other converters on its properties, which also have converters, etc.

Also we don't explain that any exception is possible since we may call user-provided converters, but from past discussions didn't attempt to explain that either.

Reply:
> but that extra verbage I thought was implied.

The fact that we state `for returnType`, makes it ambiguous. I think being explicit could be useful. Up to you :)

> Also we don't explain that any exception is possible since we may call user-provided converters, but from past discussions didn't attempt to explain that either.

Agreed.

Reply:
Is this better? I'm no writer :P

"There is no compatible JsonConverter for {0} or its serializable members."

Reply:
nice